### PR TITLE
disttask: refactor task step and batch subtask dispatching

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

- revert StepInit to -1 and StepImport to 1 to avoid potential compatibility issue
- remove StageFinished and Finished, add GetNextStep
- successfully task MUST go from StepInit to business steps, then StepDone.

since subtask metas might be too large to save in one transaction,
we plan to save subtasks in batch later, so add this:
invariant: task.Step always means the most recent step that all
corresponding subtasks have been saved to system table.

when all subtasks of task.Step is finished, we call OnNextSubtasksBatch
to generate subtasks of next step. after all subtasks of next step are
saved to system table, we will update task.Step to next step, so the
invariant hold.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/framework_dynamic_dispatch_test.go [hunk 1]
```diff
@@ -94,6 +91,7 @@ func TestFrameworkDynamicBasic(t *testing.T) {
 }
 
 func TestFrameworkDynamicHA(t *testing.T) {
+	t.Skip("skip this test because the function is not implemented yet")
```
Comment (ywqzzy) [inline disttask/framework/framework_dynamic_dispatch_test.go:94]: Why skip it?

Comment (D3Hunter) [inline disttask/framework/framework_dynamic_dispatch_test.go:94]: cc @ywqzzy

Comment (D3Hunter) [reply disttask/framework/framework_dynamic_dispatch_test.go]: reverted https://github.com/pingcap/tidb/pull/46957/commits/e573f2d2a9328371369593386ba119c764b2e281

**Review by ywqzzy — APPROVED** [review]
LGTM

**Pull request merged**